Inc/IPC/Policies/TransactionManager.h

```diff
@@ -24,17 +25,11 @@ namespace Policies
 
         explicit TransactionManager(TimeoutFactory timeoutFactory, const std::chrono::milliseconds& defaultTimeout = {})
             : m_timeoutFactory{ std::move(timeoutFactory) },
-              m_defaultTimeout{ defaultTimeout }
+              m_defaultTimeout{ defaultTimeout != std::chrono::milliseconds::zero() ? defaultTimeout : GetDefaultTimeout() }
         {}
 
         template <typename OtherContext>
-        Id BeginTransaction(OtherContext&& context)
-        {
-            return BeginTransaction(std::forward<OtherContext>(context), m_defaultTimeout);
-        }
-
-        template <typename OtherContext>
-        Id BeginTransaction(OtherContext&& context, const std::chrono::milliseconds& timeout)
+        Id BeginTransaction(OtherContext&& context, const std::chrono::milliseconds& timeout = {})

```

Comment:
Unless [duration_values::zero](http://en.cppreference.com/w/cpp/chrono/duration_values/zero) is specialized.